src/t_stream.c

```diff
@@ -3119,26 +3172,24 @@ void xclaimCommand(client *c) {
     mstime_t deliverytime = -1;  /* -1 means IDLE/TIME options not given. */
     int force = 0;
     int justid = 0;
+    int mkgroup = 0;
 
-    if (o) {
-        if (checkType(c,o,OBJ_STREAM)) return; /* Type error. */
-        group = streamLookupCG(o->ptr,c->argv[2]->ptr);
+    if(c->argc > 6 && !strcasecmp(c->argv[c->argc-1]->ptr,"MKGROUP")) {
+        mkgroup = 1;
```

Comment:
optional arguments are interchangeable, you shouldn't search for it only at the last argument.
why not move it to the argument parsing? (where you already skip that arg anyway)